integration-test/src/test/scala/com/cloudera/livy/test/BatchIT.scala

```diff
@@ -75,15 +83,80 @@ class BatchIT extends BaseIntegrationTestSuite {
   ignore("submit a SparkR application") {
     val hdfsPath = uploadResource("rtest.R")
     val result = runScript(hdfsPath)
-    assert(result.state === SessionState.Success().toString)
+    assert(result.state === SessionState.Success().toString())
+  }
+
+  test("deleting a session should kill YARN app") {
+    val output = newOutputPath()
+    val batchId = runSpark(classOf[SimpleSparkApp], List(output, "false"), waitForExit = false).id
+
+    dumpLogOnFailure(batchId, cleanup = false) {
```

Comment:
Should `cleanup` really be `false` here?

For example, if `waitUntilRunning` throws, you'll potentially leave the underlying batch session alive, which might affect subsequent tests can cause cascading failures.

The follow up question would be: does it even make sense to have `dumpLogOnFailure` have a `cleanup` parameter? Maybe it should always try to clean up the batch and ignore any errors.